src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastHelpers.cs

```diff
@@ -255,10 +278,14 @@ private static CastResult TryGet(nuint source, nuint target)
         [DebuggerStepThrough]
         private static object? IsInstanceOfClass(void* toTypeHnd, object? obj)
         {
-            if (obj == null || RuntimeHelpers.GetMethodTable(obj) == toTypeHnd)
+            if (obj == null)
+                return obj;
+
+            MethodTable* mt = RuntimeHelpers.GetMethodTable(obj);
```

Comment:
Any numbers for this change?

Reply:
It removes a `mov`, but also flips some `if`s the wrong way. Will back it out and look at it separately